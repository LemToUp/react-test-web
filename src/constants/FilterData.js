export const filterDataConstants = {
    INIT_FILTER_SECTION: 'INIT_FILTER_SECTION',
    SET_CONTEXTS_LIST_BY_FILTER: 'SET_CONTEXTS_LIST_BY_FILTER',
    SET_CONTEXTS_CHECKS_BY_FILTER: 'SET_CONTEXTS_CHECKS_BY_FILTER',
    SET_DIMENTIONS_LIST_BY_FILTER: 'SET_DIMENTIONS_LIST_BY_FILTER',
    SET_DIMENTIONS_CHECKS_BY_FILTER: 'SET_DIMENTIONS_CHECKS_BY_FILTER',
    SET_FILTERS_LIST_BY_FILTER: 'SET_FILTERS_LIST_BY_FILTER',
    SET_FILTERS_CHECKS_BY_FILTER: 'SET_FILTERS_CHECKS_BY_FILTER',
    SET_SORT_RULES_BY_FILTER: 'SET_SORT_RULES_BY_FILTER',
};