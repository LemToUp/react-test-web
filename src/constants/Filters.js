export const filtersConstants = {
    GET_CONTEXTS_LIST: 'GET_CONTEXTS_LIST',
    GET_DIMENTIONS_LIST: 'GET_DIMENTIONS_LIST',
    GET_FILTERS_LIST: 'GET_FILTERS_LIST',
};