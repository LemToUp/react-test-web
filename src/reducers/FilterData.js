"use strict";
exports.__esModule = true;
var FilterData_1 = require("../constants/FilterData");
function filterData(state, action) {
    if (state === void 0) { state = {}; }
    switch (action.type) {
        case FilterData_1.filterDataConstants.INIT_FILTER_SECTION:
            state[action.name] = state[action.name] || {};
            return Object.assign({}, state);
        case FilterData_1.filterDataConstants.SET_CONTEXTS_LIST_BY_FILTER:
            state[action.name] = Object.assign({}, state[action.name], { filterContexts: action.data });
            return Object.assign({}, state);
        case FilterData_1.filterDataConstants.SET_CONTEXTS_CHECKS_BY_FILTER:
            state[action.name] = Object.assign({}, state[action.name], { filterContextsChecks: action.data });
            return Object.assign({}, state);
        case FilterData_1.filterDataConstants.SET_DIMENTIONS_LIST_BY_FILTER_SUCCEEDED:
            state[action.name] = Object.assign({}, state[action.name], { filterDimentions: action.data });
            return Object.assign({}, state);
        case FilterData_1.filterDataConstants.SET_DIMENTIONS_CHECKS_BY_FILTER:
            state[action.name] = Object.assign({}, state[action.name], { filterDimentionsChecks: action.data });
            return Object.assign({}, state);
        case FilterData_1.filterDataConstants.SET_FILTERS_LIST_BY_FILTER_SUCCEEDED:
            state[action.name] = Object.assign({}, state[action.name], { filterFilters: action.data });
            return Object.assign({}, state);
        case FilterData_1.filterDataConstants.SET_FILTERS_CHECKS_BY_FILTER:
            state[action.name] = Object.assign({}, state[action.name], { filterFiltersChecks: action.data });
            return Object.assign({}, state);
        case FilterData_1.filterDataConstants.SET_SORT_RULES_BY_FILTER:
            state[action.name] = Object.assign({}, state[action.name], { sortRules: action.data });
            return Object.assign({}, state);
        default:
            return state;
    }
}
exports.filterData = filterData;
