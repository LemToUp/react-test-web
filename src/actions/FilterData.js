import {filterDataConstants} from '../constants/FilterData';

export const filtersDataActions = {
    initFilterSection,
    getContextsDataByFilter,
    getContextsChecksByFilter,
    setContextsDataByFilter,
    setContextsChecksByFilter,
    getDimentionsDataByFilter,
    getDimentionsChecksByFilter,
    setDimentionsDataByFilter,
    setDimentionsChecksByFilter,
    getFiltersDataByFilter,
    getFiltersChecksByFilter,
    setFiltersDataByFilter,
    setFiltersChecksByFilter,
};

function initFilterSection(name) {
    return dispatch => {
        return dispatch({type: filterDataConstants.INIT_FILTER_SECTION, name});
    }
}

function getContextsDataByFilter(name) {
    return dispatch => {
        return dispatch({type: filterDataConstants.GET_CONTEXTS_LIST_BY_FILTER, name});
    }
}

function getContextsChecksByFilter(name) {
    return dispatch => {
        return dispatch({type: filterDataConstants.GET_CONTEXTS_CHECKS_BY_FILTER, name});
    }
}

function setContextsDataByFilter(name, data) {
    return dispatch => {
        return dispatch({type: filterDataConstants.SET_CONTEXTS_LIST_BY_FILTER, name, data});
    }
}

function setContextsChecksByFilter(name, checks) {
    return dispatch => {
        return dispatch({type: filterDataConstants.SET_CONTEXTS_CHECKS_BY_FILTER, name, checks});
    }
}

function getDimentionsDataByFilter(name) {
    return dispatch => {
        return dispatch({type: filterDataConstants.GET_DIMENTIONS_LIST_BY_FILTER, name});
    };
}

function getDimentionsChecksByFilter(name) {
    return dispatch => {
        return dispatch({type: filterDataConstants.GET_DIMENTIONS_LIST_BY_FILTER, name});
    };
}

function setDimentionsDataByFilter(name, data) {
    return dispatch => {
        return dispatch({type: filterDataConstants.SET_DIMENTIONS_LIST_BY_FILTER, name, data});
    };
}

function setDimentionsChecksByFilter(name, checks) {
    return dispatch => {
        return dispatch({type: filterDataConstants.SET_DIMENTIONS_LIST_BY_FILTER, name, checks});
    };
}

function getFiltersDataByFilter(name) {
    return dispatch => {
        return dispatch({type: filterDataConstants.GET_FILTERS_LIST_BY_FILTER, name});
    };
}

function getFiltersChecksByFilter(name) {
    return dispatch => {
        return dispatch({type: filterDataConstants.GET_FILTERS_LIST_BY_FILTER, name});
    };
}

function setFiltersDataByFilter(name, data) {
    return dispatch => {
        return dispatch({type: filterDataConstants.SET_FILTERS_LIST_BY_FILTER, name, data});
    };
}

function setFiltersChecksByFilter(name, checks) {
    return dispatch => {
        return dispatch({type: filterDataConstants.SET_FILTERS_LIST_BY_FILTER, name, checks});
    };
}