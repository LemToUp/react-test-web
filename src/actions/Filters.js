import { filtersConstants } from '../constants/Filters';
import  { filtersService } from '../services/Filters'

export const filtersActions = {
    getContexts,
    getDimentions,
    getFilters,
};

function getContexts() {
    return {type: filtersConstants.CONTEXTS_LIST_REQUESTED}
    /*return dispatch => {
        return filtersService.getContextsList().then((data)=> {
            return dispatch({type: filtersConstants.GET_CONTEXTS_LIST, data});
        });
    }*/
}

function setContexts() {
    return {type: filtersConstants.GET_CONTEXTS_LIST}
    /*return dispatch => {
        return filtersService.getContextsList().then((data)=> {
            return dispatch({type: filtersConstants.GET_CONTEXTS_LIST, data});
        });
    }*/
}

function getDimentions(ids) {
    return dispatch => {
        return filtersService.getDimentionsList(ids).then((data)=> {
            return dispatch({type: filtersConstants.GET_DIMENTIONS_LIST, data});
        });
    }
}

function getFilters(ids, filters) {
    return dispatch => {
        return filtersService.getFiltersList(ids, filters).then((data)=> {
            return dispatch({type: filtersConstants.GET_FILTERS_LIST, data});
        });
    }
}