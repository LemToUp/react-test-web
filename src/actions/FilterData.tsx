import {filterDataConstants} from '../constants/FilterData';

const initFilterSection = (name:string) => ({type: filterDataConstants.INIT_FILTER_SECTION, name});

const setContextsDataByFilter = (name:string, data:object) => ({type: filterDataConstants.SET_CONTEXTS_LIST_BY_FILTER, name, data});
const setContextsChecksByFilter = (name:string, data:object) => ({type: filterDataConstants.SET_CONTEXTS_CHECKS_BY_FILTER, name, data});

const setDimentionsDataByFilter = (name:string, data:object) => ({type: filterDataConstants.SET_DIMENTIONS_LIST_BY_FILTER, name, data});
const setDimentionsChecksByFilter = (name:string, data:object) => ({type: filterDataConstants.SET_DIMENTIONS_CHECKS_BY_FILTER, name, data});

const setFiltersDataByFilter = (name:string, data:object) => ({type: filterDataConstants.SET_FILTERS_LIST_BY_FILTER, name, data});
const setFiltersChecksByFilter = (name:string, data:object) => ({type: filterDataConstants.SET_FILTERS_CHECKS_BY_FILTER, name, data});

const setSortRulesByFilter = (name:string, data:object) => ({type: filterDataConstants.SET_SORT_RULES_BY_FILTER, name, data});

export const filtersDataActions = {
    initFilterSection,
    setContextsDataByFilter,
    setContextsChecksByFilter,
    setDimentionsDataByFilter,
    setDimentionsChecksByFilter,
    setFiltersDataByFilter,
    setFiltersChecksByFilter,
    setSortRulesByFilter,
};