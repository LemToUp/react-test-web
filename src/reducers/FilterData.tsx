import { filterDataConstants } from '../constants/FilterData';

export function filterData(state = {}, action: {type:string ,name:string , data:object}) {
    switch (action.type) {
        case filterDataConstants.INIT_FILTER_SECTION:
            state[action.name] = state[action.name] || {};
            return Object.assign({}, state);

        case filterDataConstants.SET_CONTEXTS_LIST_BY_FILTER:
            state[action.name] = Object.assign({}, state[action.name], {filterContexts: action.data});
            return Object.assign({}, state);
        case filterDataConstants.SET_CONTEXTS_CHECKS_BY_FILTER:
            state[action.name] = Object.assign({}, state[action.name], {filterContextsChecks: action.data});
            return Object.assign({}, state);

        case filterDataConstants.SET_DIMENTIONS_LIST_BY_FILTER_SUCCEEDED:
            state[action.name] = Object.assign({}, state[action.name], {filterDimentions: action.data});
            return Object.assign({}, state);
        case filterDataConstants.SET_DIMENTIONS_CHECKS_BY_FILTER:
            state[action.name] = Object.assign({}, state[action.name], {filterDimentionsChecks: action.data});
            return Object.assign({}, state);

        case filterDataConstants.SET_FILTERS_LIST_BY_FILTER_SUCCEEDED:
            state[action.name] = Object.assign({}, state[action.name], {filterFilters: action.data});
            return Object.assign({}, state);
        case filterDataConstants.SET_FILTERS_CHECKS_BY_FILTER:
            state[action.name] = Object.assign({}, state[action.name], {filterFiltersChecks: action.data});
            return Object.assign({}, state);

        case filterDataConstants.SET_SORT_RULES_BY_FILTER:
            state[action.name] = Object.assign({}, state[action.name], {sortRules: action.data});
            return Object.assign({}, state);
        default:
            return state;
    }
}