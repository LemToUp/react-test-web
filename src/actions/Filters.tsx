import {filtersConstants} from '../constants/Filters';

const getContexts = () => ({type: filtersConstants.GET_CONTEXTS_LIST});

export const filtersActions = {
    getContexts,
};