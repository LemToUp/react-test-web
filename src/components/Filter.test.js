import React from 'react';
import ReactDOM from 'react-dom';
import ConnectedFilter, {Filter} from './Filter';
import {Provider} from 'react-redux';
import configureStore from '../store/ConfigureStore';

import {filtersActions} from '../actions/Filters';
import {filtersDataActions} from '../actions/FilterData';
import {mount, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

const store = configureStore();

const filterName = 'filter_test';
store.dispatch(filtersDataActions.initFilterSection(filterName));

it('renders without crashing', () => {
    return store.dispatch(filtersActions.getContexts()).then(action => {
        expect(action.data.length).toBe(2);

        const div = document.createElement('div');
        ReactDOM.render(
            <Provider store={store}>
                <ConnectedFilter
                    name={`filter_test`}
                />
            </Provider>
            , div);
        ReactDOM.unmountComponentAtNode(div);
        return action;
    });
});


describe('+Module testing of Filter and Redux', () => {
    let store, wrapper;
    configure({adapter: new Adapter()});
    beforeEach(() => {
        store = configureStore();
        store.dispatch(filtersDataActions.initFilterSection(filterName));
    });
    it('++ check Data logic', () => {
        return new Promise(resolve => {
            return resolve(store.dispatch(filtersActions.getContexts()));
        }).then(() => {
            wrapper = mount(
                <Provider store={store}>
                    <ConnectedFilter
                        name={`filter_test`}
                    />
                </Provider>);
            const filter = wrapper.find(Filter);
            expect(filter.prop('contexts').length).toEqual(2);
            filter.prop('setContextsChecks')([2]);
            filter.update();

            /*wrapper.find(Filter).prop('getDimentionsFromState')([1,2]).then(action => {
                filter.prop('setDimentions')(action.data);
                return action;
            });*/
        });
    });
});
