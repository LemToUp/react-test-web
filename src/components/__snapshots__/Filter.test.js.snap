// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select contexts 1`] = `
<div
  className="Filter-modal-wrapper"
>
  <div
    className="Filter-modal container"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="Filter-modal-header row p-1"
    >
      <div
        className="col-2 px-1 pt-1"
      >
        <i
          className="material-icons draggable"
          onMouseDown={[Function]}
        >
          drag_indicator
        </i>
      </div>
      <span
        className="col-8"
      >
        FILTERS
      </span>
      <span
        className="Filter-close-icon col-2 pt-1 text-right"
      >
        <i
          className="material-icons pointer"
          onClick={[Function]}
        >
          close
        </i>
      </span>
    </div>
    <div
      className="Filter-modal-body row align-items-end"
    >
      <div
        className="Filter-section Drop-down-section col-md-10 offset-md-2 p-1"
      >
        <p>
          <i
            className="material-icons pointer p-1"
            onClick={[Function]}
          >
            expand_more
          </i>
          <span
            className="Drop-down-section-title mr-2"
          >
            CONTEXTS
          </span>
          <span
            className="Drop-down-section-row"
          >
            
          </span>
        </p>
        <div
          className="Drop-down-modal d-none"
        />
      </div>
      <div
        className="Filter-section Drop-down-section col-md-10 offset-md-2 p-1"
      >
        <p>
          <i
            className="material-icons pointer p-1"
            onClick={[Function]}
          >
            expand_more
          </i>
          <span
            className="Drop-down-section-title mr-2"
          >
            DIMENTIONS
          </span>
          <span
            className="Drop-down-section-row"
          >
            
          </span>
        </p>
        <div
          className="Drop-down-modal d-none"
        />
      </div>
      <div
        className="Filter-section Search-filter-section col-md-10 offset-md-2 p-2"
      >
        <div
          className="input-group mb-1"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text p-1"
            >
              <i
                className="material-icons pointer"
                onClick={[Function]}
              >
                search
              </i>
            </span>
          </div>
          <input
            className="form-control p-1"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
          />
        </div>
        <div
          aria-label="..."
          className="btn-group btn-group-sm"
          role="group"
        >
          <button
            className="btn btn-secondary "
            onClick={[Function]}
            type="button"
          >
            A-Z
          </button>
          <div
            className="btn-group btn-group-sm"
            role="group"
          >
            <button
              className="btn btn-secondary"
              onClick={[Function]}
              type="button"
            >
              
            </button>
            <div
              aria-labelledby="btnGroupDrop1"
              className="dropdown-menu mt-0"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <button
                className="dropdown-item py-1 px-2"
                onClick={[Function]}
              >
                **
              </button>
              <button
                className="dropdown-item py-1 px-2"
                onClick={[Function]}
              >
                *_
              </button>
              <button
                className="dropdown-item py-1 px-2"
                onClick={[Function]}
              >
                ""
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="Filter-section Content-filter-section col-md-10 offset-md-2 p-1"
      />
    </div>
  </div>
</div>
`;

exports[`Select contexts 2`] = `
<div
  className="Filter-modal-wrapper"
>
  <div
    className="Filter-modal container"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="Filter-modal-header row p-1"
    >
      <div
        className="col-2 px-1 pt-1"
      >
        <i
          className="material-icons draggable"
          onMouseDown={[Function]}
        >
          drag_indicator
        </i>
      </div>
      <span
        className="col-8"
      >
        FILTERS
      </span>
      <span
        className="Filter-close-icon col-2 pt-1 text-right"
      >
        <i
          className="material-icons pointer"
          onClick={[Function]}
        >
          close
        </i>
      </span>
    </div>
    <div
      className="Filter-modal-body row align-items-end"
    >
      <div
        className="Filter-section Drop-down-section col-md-10 offset-md-2 p-1"
      >
        <p>
          <i
            className="material-icons pointer p-1"
            onClick={[Function]}
          >
            expand_more
          </i>
          <span
            className="Drop-down-section-title mr-2"
          >
            CONTEXTS
          </span>
          <span
            className="Drop-down-section-row"
          >
            
          </span>
        </p>
        <div
          className="Drop-down-modal d-none"
        >
          <ul
            className="list-group pr-5"
          >
            <li
              className="p-1"
            >
              <div
                className="form-check"
              >
                <input
                  className="form-check-input"
                  id="0.6295889167233539_0"
                  name={1}
                  onChange={[Function]}
                  type="checkbox"
                  value={1}
                />
                <label
                  className="form-check-label"
                  htmlFor="0.6295889167233539_0"
                >
                  Test Story
                </label>
              </div>
            </li>
            <li
              className="p-1"
            >
              <div
                className="form-check"
              >
                <input
                  className="form-check-input"
                  id="0.6295889167233539_1"
                  name={2}
                  onChange={[Function]}
                  type="checkbox"
                  value={2}
                />
                <label
                  className="form-check-label"
                  htmlFor="0.6295889167233539_1"
                >
                  Editor
                </label>
              </div>
            </li>
          </ul>
        </div>
      </div>
      <div
        className="Filter-section Drop-down-section col-md-10 offset-md-2 p-1"
      >
        <p>
          <i
            className="material-icons pointer p-1"
            onClick={[Function]}
          >
            expand_more
          </i>
          <span
            className="Drop-down-section-title mr-2"
          >
            DIMENTIONS
          </span>
          <span
            className="Drop-down-section-row"
          >
            
          </span>
        </p>
        <div
          className="Drop-down-modal d-none"
        />
      </div>
      <div
        className="Filter-section Search-filter-section col-md-10 offset-md-2 p-2"
      >
        <div
          className="input-group mb-1"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text p-1"
            >
              <i
                className="material-icons pointer"
                onClick={[Function]}
              >
                search
              </i>
            </span>
          </div>
          <input
            className="form-control p-1"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
          />
        </div>
        <div
          aria-label="..."
          className="btn-group btn-group-sm"
          role="group"
        >
          <button
            className="btn btn-secondary "
            onClick={[Function]}
            type="button"
          >
            A-Z
          </button>
          <div
            className="btn-group btn-group-sm"
            role="group"
          >
            <button
              className="btn btn-secondary"
              onClick={[Function]}
              type="button"
            >
              **
            </button>
            <div
              aria-labelledby="btnGroupDrop1"
              className="dropdown-menu mt-0"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <button
                className="dropdown-item py-1 px-2"
                onClick={[Function]}
              >
                **
              </button>
              <button
                className="dropdown-item py-1 px-2"
                onClick={[Function]}
              >
                *_
              </button>
              <button
                className="dropdown-item py-1 px-2"
                onClick={[Function]}
              >
                ""
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="Filter-section Content-filter-section col-md-10 offset-md-2 p-1"
      />
    </div>
  </div>
</div>
`;

exports[`renders without crashing 1`] = `
<div
  className="Filter-modal-wrapper"
>
  <div
    className="Filter-modal container"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="Filter-modal-header row p-1"
    >
      <div
        className="col-2 px-1 pt-1"
      >
        <i
          className="material-icons draggable"
          onMouseDown={[Function]}
        >
          drag_indicator
        </i>
      </div>
      <span
        className="col-8"
      >
        FILTERS
      </span>
      <span
        className="Filter-close-icon col-2 pt-1 text-right"
      >
        <i
          className="material-icons pointer"
          onClick={[Function]}
        >
          close
        </i>
      </span>
    </div>
    <div
      className="Filter-modal-body row align-items-end"
    >
      <div
        className="Filter-section Drop-down-section col-md-10 offset-md-2 p-1"
      >
        <p>
          <i
            className="material-icons pointer p-1"
            onClick={[Function]}
          >
            expand_more
          </i>
          <span
            className="Drop-down-section-title mr-2"
          >
            CONTEXTS
          </span>
          <span
            className="Drop-down-section-row"
          >
            
          </span>
        </p>
        <div
          className="Drop-down-modal d-none"
        >
          <ul
            className="list-group pr-5"
          >
            <li
              className="p-1"
            >
              <div
                className="form-check"
              >
                <input
                  className="form-check-input"
                  id="0.33806233188690493_0"
                  name={1}
                  onChange={[Function]}
                  type="checkbox"
                  value={1}
                />
                <label
                  className="form-check-label"
                  htmlFor="0.33806233188690493_0"
                >
                  Test Story
                </label>
              </div>
            </li>
            <li
              className="p-1"
            >
              <div
                className="form-check"
              >
                <input
                  className="form-check-input"
                  id="0.33806233188690493_1"
                  name={2}
                  onChange={[Function]}
                  type="checkbox"
                  value={2}
                />
                <label
                  className="form-check-label"
                  htmlFor="0.33806233188690493_1"
                >
                  Editor
                </label>
              </div>
            </li>
          </ul>
        </div>
      </div>
      <div
        className="Filter-section Drop-down-section col-md-10 offset-md-2 p-1"
      >
        <p>
          <i
            className="material-icons pointer p-1"
            onClick={[Function]}
          >
            expand_more
          </i>
          <span
            className="Drop-down-section-title mr-2"
          >
            DIMENTIONS
          </span>
          <span
            className="Drop-down-section-row"
          >
            
          </span>
        </p>
        <div
          className="Drop-down-modal d-none"
        />
      </div>
      <div
        className="Filter-section Search-filter-section col-md-10 offset-md-2 p-2"
      >
        <div
          className="input-group mb-1"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text p-1"
            >
              <i
                className="material-icons pointer"
                onClick={[Function]}
              >
                search
              </i>
            </span>
          </div>
          <input
            className="form-control p-1"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
          />
        </div>
        <div
          aria-label="..."
          className="btn-group btn-group-sm"
          role="group"
        >
          <button
            className="btn btn-secondary "
            onClick={[Function]}
            type="button"
          >
            A-Z
          </button>
          <div
            className="btn-group btn-group-sm"
            role="group"
          >
            <button
              className="btn btn-secondary"
              onClick={[Function]}
              type="button"
            >
              **
            </button>
            <div
              aria-labelledby="btnGroupDrop1"
              className="dropdown-menu mt-0"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <button
                className="dropdown-item py-1 px-2"
                onClick={[Function]}
              >
                **
              </button>
              <button
                className="dropdown-item py-1 px-2"
                onClick={[Function]}
              >
                *_
              </button>
              <button
                className="dropdown-item py-1 px-2"
                onClick={[Function]}
              >
                ""
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="Filter-section Content-filter-section col-md-10 offset-md-2 p-1"
      />
    </div>
  </div>
</div>
`;
